src/samples/BoundingBox.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Rock-Core
+#ifndef BASE_SAMPLES_BOUNDINGBOX_HPP_
+#define BASE_SAMPLES_BOUNDINGBOX_HPP_
+
+#include <base/Eigen.hpp>
+#include <base/Time.hpp>
+
+namespace base {
+namespace samples {
+/**
+ * Axis-aligned bounding box (ABB) representation with associated uncertainty
+ * information.
+ */
+struct BoundingBox {
+    /* The sample timestamp. */
+    base::Time time;
+
+    /* Bounding box center position. */
+    base::Vector3d position;
+
+    /* Bounding box dimension. */
+    base::Vector3d dimension;
+
+    /* Covariance matrix of the position. */
+    base::Matrix3d cov_position;
+
+    /* Covariance matrix of the dimension. */
+    base::Matrix3d cov_dimension;
+
+    BoundingBox()
+        : time(),
+          position(base::Vector3d(0, 0, 0)),
```

Comment:
The convention so far for pre-initialized types is to always use `base::unknown` (a.k.a. NAN)